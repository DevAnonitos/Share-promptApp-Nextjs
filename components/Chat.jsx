import Widget from 'rasa-webchat';

const Chat = () => {
    return (
        <Widget
            initPayload={"/greet"}
            socketUrl={"http://localhost:5005"}
            socketPath={"/socket.io/"}
            customData={{"language": "en"}}
            title={"BotChat"}
            inputTextFieldHint={"Enter to chat with bot..."}
        />
    );
};

export default Chat;
