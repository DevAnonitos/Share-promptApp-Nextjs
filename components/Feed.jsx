"use-client";

import React, { useState, useEffect } from 'react';
import PromptCard from "./PromptCard";
import { findDOMNode } from 'react-dom';

const PromptCardList = ({ data, handleTagClick }) => {
    return (
        <div className='mt-16 prompt_layout'>
            {data.map((post) => (
                <PromptCard
                    key={post._id}
                    post={post}
                    handleTagClick={handleTagClick}
                />
            ))}
        </div>
    );
};

const Feed = () => {
    // Fetch AllPosts
    const [allPosts, setAllPosts] = useState([]);

    const [searchText, setSearchText] = useState("");
    const [searchedResults, setSearchedResults] = useState([]);
    const [searchTimeout, setSearchTimeout] = useState(null);

    // Handle Logic state
    const fetchPosts = async () => {
        const response = await fetch("/api/prompt");
        const data = await response.json();

        setAllPosts(data);
    };

    useEffect(() => {
        fetchPosts();
    },[]);

    const filterPrompt = (searchtext) => {
        const regex = new RegExp(searchtext, "i");
        return allPosts.filter(
            (item) =>
                regex.test(item.creator.username) ||
                regex.test(item.tag) ||
                regex.test(item.prompt)
        );
    };

    const handleSearchChange = (e) => {

    };

    const handleTagClick = (tagName) => {
        setSearchText(tagName);
    };

    return (
        <section className='feed'>
            <form className='relative w-full flex-center'>
                <input
                    type='text'
                    placeholder='Search for a tag or a username...'
                    value={searchText}
                    required
                    onChange={handleSearchChange}
                    className='search_input peer'
                />
            </form>
            {/* Prompt Card list */}
            {searchText ? (
                <>
                    <PromptCardList
                        data={searchedResults}
                        handleTagClick={handleTagClick}
                    />
                </>
            ) : (
                <>
                    <PromptCardList
                        data={allPosts}
                        handleTagClick={handleTagClick}
                    />
                </>
            )}
        </section>
    );
};

export default Feed;
