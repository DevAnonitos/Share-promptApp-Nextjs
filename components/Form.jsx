"use client"

import Link from "next/link";
import React, { useState, useEffect } from 'react';

const Form = ({ type, post, setPost, submitting, handleSubmit }) => {

    const [placeholderIndex, setPlaceholderIndex] = useState(0);

    const placeholder = "Write Your Prompt Here...";
    const typingSpeed = 250;

    useEffect(() => {
        const interval = setInterval(() => {
        setPlaceholderIndex((index) =>
            index >= placeholder.length ? 0 : index + 1
        );
        }, typingSpeed);
        return () => clearInterval(interval);
    }, []);

    return (
        <>
            <section
                className="w-full max-w-full flex-center flex-col"
            >
                <h1 className="head_text text-center">
                    <span className="text-gradient-alt animate-gradient">{type} Post </span>
                </h1>
                <p className="desc text-center max-w-md">
                    {type} and share amazing prompts with the world, and let your
                    imagination run wild with any AI-powered platform
                </p>

                <form
                    onSubmit={handleSubmit}
                    className="mt-10 w-full max-w-2xl
                    flex flex-col gap-7 glassmorphism shadow-xl"
                >
                    <label>
                        <span
                            className="font-satoshi font-semibold text-base text-gray-700"
                        >
                            Your AI Prompt
                            <textarea
                                value={post.prompt}
                                onChange={(e) =>
                                setPost({
                                        ...post,
                                        prompt: e.target.value,
                                    })
                                }
                                placeholder={placeholder.substring(0, placeholderIndex)}
                                required
                                className="form_textarea"
                                onKeyPress={() => setPlaceholderIndex(0)}
                                onFocus={() => setPlaceholderIndex(0)}
                                onBlur={() => setPlaceholderIndex(0)}
                            ></textarea>
                        </span>
                    </label>

                    <label>
                        <span
                            className='font-satoshi font-semibold text-base text-gray-700'
                        >
                            Field of Prompt{" "}
                            <span className='font-normal'>
                                (#Product, #Webdevelopment, #Idea, etc.)
                            </span>
                        </span>
                        <input
                            value={post.tag}
                            onChange={(e) =>
                                setPost({
                                    ...post,
                                    tag: e.target.value
                                }
                            )}
                            type='text'
                            placeholder='#Tag'
                            required
                            className='form_input'
                        />
                    </label>

                    {/* Button */}
                    <div className="flex-end mx-3 mb-5 gap-4">
                        <Link
                            href="/"
                            className="text-gray-500 text-sm"
                        >
                            Cancel
                        </Link>
                        <button
                            type="submit"
                            disabled={submitting}
                            className="px-5 py-1.5 text-sm
                            bg-primary-orange rounded-full text-white"
                        >
                            {submitting ? `${type}ing...` : type}
                        </button>
                    </div>
                </form>
            </section>
        </>
    );
};

export default Form;
