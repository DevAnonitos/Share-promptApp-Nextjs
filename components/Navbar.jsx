"use client"

import React, { useState, useEffect } from 'react';
import Link from 'next/link';
import Image from 'next/image';
import { signIn, signOut, useSession, getProviders } from "next-auth/react";

const Navbar = () => {

    const isUserLoggedIn = true;

    const [toggleDropdown, setToggleDropdown] = useState(false);
    // const [providers, setProviders] = useState(null);

    // useEffect(() => {
    //     (
    //         async () => {
    //             const res = await getProviders();
    //             setProviders(res);
    //         }
    //     )();
    // }, []);

    return (
        <>
            <nav className='flex items-center flex-between w-full mb-16 pt-3'>
                <Link href="/" className='flex gap-2 flex-center'>
                    <Image
                        src="./assets/images/logo.svg"
                        alt='logo'
                        width={30}
                        height={30}
                        className='object-contain'
                    />
                    <p className='logo_text'>PromptApp</p>
                </Link>

                {/* Desktop navigation */}
                <div className='sm:flex hidden'>
                    {isUserLoggedIn ? (
                        <div className='flex gap-3 md:gap-5'>
                            <Link
                                href="/create-prompt"
                                className='black_btn'
                            >
                                Create Post
                            </Link>

                            <button
                                type='button'
                                // onClick={signOut}
                                className='outline_btn'
                            >
                                SignOut
                            </button>

                            <Link href="/profile">
                                <Image
                                    src="/assets/images/logo.svg"
                                    width={37}
                                    height={37}
                                    className='rounded-full ring-2 bottom-2'
                                    alt='profile'
                                />
                            </Link>
                        </div>
                    ) : (
                        <>
                            {/* {providers && Object.values(providers).map((provider) => (
                                <button
                                    type='button'
                                    key={provider.name}
                                    onClick={() => {
                                        signIn(provider.id)
                                    }}
                                    className='black_btn'
                                >
                                    SignIn
                                </button>
                            ))} */}
                        </>
                    )}
                </div>

                {/* Mobile navigation */}
                <div className='sm:hidden flex relative'>
                    {isUserLoggedIn ? (
                        <div className='flex'>
                            <Image
                                src="/assets/images/logo.svg"
                                width={37}
                                height={37}
                                className='rounded-full ring-2 bottom-2 cursor-pointer'
                                alt='profile'
                                onClick={() => setToggleDropdown(!toggleDropdown)}
                            />
                            {/* Toggle DropDown */}

                            {toggleDropdown && (
                                <div className='dropdown z-50'>
                                    <Link
                                        href="/profile"
                                        className='dropdown_link border-animation '
                                        onClick={() => setToggleDropdown(false)}
                                    >
                                        My Profile
                                    </Link>
                                    <Link
                                        href="/create-prompt"
                                        className='dropdown_link border-animation '
                                        onClick={() => setToggleDropdown(false)}
                                    >
                                        Create Prompt
                                    </Link>
                                    <button
                                        type='button'
                                        onClick={() => {
                                            setToggleDropdown(false);
                                            signOut();
                                        }}
                                        className='mt-5 w-full black_btn'
                                    >
                                        SignOut
                                    </button>
                                </div>
                            )}
                        </div>
                    ) : (
                        <>

                        </>
                    )}
                </div>
            </nav>
        </>
    );
};

export default Navbar;
