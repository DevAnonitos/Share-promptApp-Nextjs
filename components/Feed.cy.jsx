import React from 'react'
import Feed from './Feed'

describe('<Feed />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-react
    cy.mount(<Feed />)
  })
})