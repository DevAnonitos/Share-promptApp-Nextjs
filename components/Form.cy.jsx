import React from 'react'
import Form from './Form'

describe('<Form />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-react
    cy.mount(<Form />)
  })
})