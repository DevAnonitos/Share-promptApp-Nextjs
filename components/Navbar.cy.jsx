import React from 'react'
import Navbar from './Navbar'

describe('<Navbar />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-react
    cy.mount(<Navbar />)
  })
})