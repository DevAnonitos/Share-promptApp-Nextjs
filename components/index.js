import Feed from "./Feed";
import Form from "./Form";
import Navbar from "./Navbar";
import Profile from "./Profile";
import PromptCard from "./PromptCard";
import Provider from "./Provider";

// export All Components
export {
    Feed,
    Form,
    Navbar,
    Profile,
    PromptCard,
    Provider,
};
