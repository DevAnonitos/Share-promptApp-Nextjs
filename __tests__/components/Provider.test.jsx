import { Provider } from "@/components";
import React from "react";
import { render, screen, waitFor } from "@testing-library/react";

describe("ProviderComponent", () => {
  it("should render Provider Components", () => {
    expect(Provider).toBeDefined()
    expect(Provider).toMatchSnapshot()
  })
})