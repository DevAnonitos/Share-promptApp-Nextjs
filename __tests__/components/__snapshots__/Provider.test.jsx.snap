// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderComponent should render Provider Components 1`] = `[Function]`;
